app/src/main/java/de/triology/cas/services/RegistryEtcd.java

```diff
@@ -199,4 +246,89 @@ protected JSONObject getCurrentDoguNode(String doguName) throws ParseException {
         }
         return json;
     }
+
+    /**
+     * sanitize the clientId.
+     * <p>
+     * The clientId should only consists out of letters and numbers, no whitespace or other special characters.
+     * We sanitize the clientId because we call doguctl in getCurrentOAuthClientSecret() with the clientId as an argument.
+     *
+     * @param clientID
+     * @return the clientId if everything is ok, otherwise an empty string
+     */
+    private String sanitizeClientID(String clientID) throws IllegalArgumentException {
+        if (!clientID.matches("^[a-zA-Z0-9_]*$")) {
+            // clientId is "unclean"
+            return "";
+        }
+        return clientID;
+    }
+
+
+    /**
+     * Retrieves the client secret for a given clientId and encrypts the secret accordingly.
+     *
+     * @param clientID Identifier for the OAuth client
+     * @return hash of the actual client secret
+     * <p>
+     */
+    protected String getCurrentOAuthClientSecret(String clientID) {
+
```

Comment:
We currently only save the hash of the password non decrypted in the etcd. This is open for discussion.

Reply:
New ticket will be created.